spec.html

```diff
@@ -24753,8 +24753,8 @@ <h1>Runtime Semantics: CreateDynamicFunction ( _constructor_, _newTarget_, _kind
             1. NOTE: Async functions are not constructable and do not have a [[Construct]] internal method or a `"prototype"` property.
             1. Perform SetFunctionName(_F_, `"anonymous"`).
             1. Let _prefix_ be the prefix associated with _kind_ in <emu-xref href="#table-dynamic-function-sourcetext-prefixes"></emu-xref>.
-            1. Let _sourceText_ be the string-concatenation of _prefix_, `" anonymous("`, _P_, 0x000A (LINE FEED), `") {"`, 0x000A (LINE FEED), _bodyText_, 0x000A (LINE FEED), and `"}"`.
-            1. Set _F_.[[SourceText]] to _sourceText_.
+            1. Let _sourceString_ be the string-concatenation of _prefix_, `" anonymous("`, _P_, 0x000A (LINE FEED), `") {"`, 0x000A (LINE FEED), _bodyText_, 0x000A (LINE FEED), and `"}"`.
+            1. Set _F_.[[SourceText]] to the result of applying UTF-16 decoding to _sourceString_.
```

Comment:
How does this work? Normally you decode bytes.

Reply:
That's because "normally" you're decoding via UTF-8, whose code unit is 8 bits. For UTF-16 the code unit is 16 bits.

But really, "applying UTF-16 decoding" (which phrase is already used elsewhere in the spec) is meant to evoke ECMAScript's version of UTF-16 decoding, described in [6.1.4 The String Type](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-string-type).  In my opinion, there should be an abstract operation, but that's a different PR.

Reply:
No, normally you're (text) decoding bytes, whether they are UTF-8 or UTF-16 bytes is a separate manner. Decoding 16-bit units isn't really a thing, but I guess it vaguely is here then. And indeed, an abstract operation (with a better name) would go a long way.